storm-client/src/jvm/org/apache/storm/spout/ShellSpout.java

```diff
@@ -177,6 +179,7 @@ private void handleMetrics(ShellMsg shellMsg) {
     private void querySubprocess() {
         try {
             markWaitingSubprocess();
+            _logHandler.setUpContext(_process, _context);
```

Comment:
This can be on-time setup, and maybe better to move this to `open()`, since we are more familiar with initializing everything in it.